### ISSUE TITLE
Terraform plans with the -json option do not log a message of the "output" type

### ISSUE THREAD

**Issue by pballandras** [hashicorp/terraform#29910]
### Terraform Version

```
bash-5.1# terraform --version
Terraform v1.0.2
on linux_amd64
```

### Terraform Configuration Files

```terraform
# _output.tf
output "hello-world" {
  value = "hello maintainers!"
}
```

### Debug Output

The problem I'm about to list is more of an incorrect behaviour as per the documentation and I think the debug logs would be a lot for something like this. Feel free to tell me I'm wrong here and I'll provide the traces.
Here's the not debug output:
```shell
bash-5.1# terraform plan -json
{"@level":"info","@message":"Terraform 1.0.2","@module":"terraform.ui","@timestamp":"2021-11-09T16:44:43.903582Z","terraform":"1.0.2","type":"version","ui":"0.1.0"}
{"@level":"info","@message":"Plan: 0 to add, 0 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"2021-11-09T16:44:45.837810Z","changes":{"add":0,"change":0,"remove":0,"operation":"plan"},"type":"change_summary"}
```
### Expected Behavior

In the documentation on the _new_ Machine-Readable UI [here](https://www.terraform.io/docs/internals/machine-readable-ui.html#outputs-1), it is says that 
> After a successful plan or apply, a message with type outputs contains the values of all root module output values. 

So calling `terraform plan -json` should output a json formatted message containing the outputs. Like this (after the changes I did):
```shell
bash-5.1# terraform plan -json
{"@level":"info","@message":"Terraform 1.1.0-dev","@module":"terraform.ui","@timestamp":"2021-11-09T20:44:12.995602Z","terraform":"1.1.0-dev","type":"version","ui":"1.0"}
{"@level":"info","@message":"Plan: 0 to add, 0 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"2021-11-09T20:44:14.940432Z","changes":{"add":0,"change":0,"remove":0,"operation":"plan"},"type":"change_summary"}
{"@level":"info","@message":"Outputs: 3","@module":"terraform.ui","@timestamp":"2021-11-09T20:44:14.940544Z","outputs":{"hello-world":{"sensitive":false,"action":"create"}},"type":"outputs"}
```
### Actual Behavior

The `-json` flag works wonderfully in the `apply`, but no message of type `outputs` appears after the `plan`.
### Steps to Reproduce

1. `terraform init`
2. `terraform plan -json` # no message of output type
3. `terraform apply -json` # very nice message of output type
### Additional Context

I isolated the Terraform executable from our wrapping system so that only variables (`var.thing` for example) were included and ran with the only the `_outputs.tf` file as written above.

Also, I looked through the code and found that there's a difference between the `ApplyCommand.Run(...)` and `PlanCommand.Run(...)` and now, before anything, of course there are differences between two completely different commands but, [there's a call in `ApplyCommand.Run(...)`](https://github.com/hashicorp/terraform/blob/21750037d5a2ed7a0a20112ee11b029445bbd1d1/internal/command/apply.go#L137)
```golang
func (c *ApplyCommand) Run(rawArgs []string) int {
    // snip ...
	// Render the resource count and outputs, unless those counts are being
	// rendered already in a remote Terraform process.
	if rb, isRemoteBackend := be.(BackendWithRemoteTerraformVersion); !isRemoteBackend || rb.IsLocalOperations() {
		view.ResourceCount(args.State.StateOutPath)
		if !c.Destroy && op.State != nil {
			view.Outputs(op.State.RootModule().OutputValues)
		}
	}
    // snip ...
```
that doesn't exist in the `PlanCommand`. I think that adding something similar to this to the `PlanCommand.Run(...)` would probably do the trick.

I'd be happy to test it out and open a PR if that's something you think could work.

### **Update**:
I patched something up and got the behaviour that (at least I think) is intended! If that would interest you guys, I'll open a PR and let you guys review it.

**Comment by crw** [comment]
Hi @pballandras, thank you for the detailed report! When I run version 1.0.11 (latest) against your `_output.tf` file as described in the issue report, I see the correct output per your description:

```
 % terraform plan -json
{"@level":"info","@message":"Terraform 1.0.11","@module":"terraform.ui","@timestamp":"2021-11-11T15:47:09.772751-08:00","terraform":"1.0.11","type":"version","ui":"0.1.0"}
{"@level":"info","@message":"Plan: 0 to add, 0 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"2021-11-11T15:47:09.793906-08:00","changes":{"add":0,"change":0,"remove":0,"operation":"plan"},"type":"change_summary"}
{"@level":"info","@message":"Outputs: 1","@module":"terraform.ui","@timestamp":"2021-11-11T15:47:09.793933-08:00","outputs":{"hello-world":{"sensitive":false,"action":"create"}},"type":"outputs"}
```

When I run 1.0.2, I get the same results you do. I did a little digging and this was fixed in 1.0.5. See the release notes for a link to the pull request with the fix: https://github.com/hashicorp/terraform/releases/tag/v1.0.5

**Comment by pballandras** [comment]
I was 3 patches away from the fix 😅. Thanks for the answer!
